src/pages/commits/ActiveCommits/BuildVariantCard.tsx

```diff
@@ -67,34 +89,52 @@ const RenderGroupedIcons = ({ tasks }) => {
   );
   return (
     <>
-      {otherTasks.map(({ count, umbrellaStatus }) => (
-        <GroupedTaskStatusBadgeWrapper>
-          <GroupedTaskStatusBadge
-            status={umbrellaStatus}
-            key={`${umbrellaStatus}_groupedBadge`}
-            count={count}
-          />
-        </GroupedTaskStatusBadgeWrapper>
-      ))}
+      {otherTasks.map(({ count, umbrellaStatus }) => {
+        const onClick = () =>
+          push(
+            getVersionRoute(versionId, {
+              statuses: mapUmbrellaStatusToQueryParam[umbrellaStatus],
+              variant: buildVariantDisplayName,
+            })
+          );
+        return (
+          <GroupedTaskStatusBadgeWrapper>
+            <GroupedTaskStatusBadge
+              status={umbrellaStatus}
+              key={`${umbrellaStatus}_groupedBadge`}
+              count={count}
+              onClick={onClick}
+            />
+          </GroupedTaskStatusBadgeWrapper>
+        );
+      })}
     </>
   );
 };
 
-const RenderTaskIcons = ({ tasks }) => (
-  <>
-    {tasks.map(({ id, status }) => (
-      <IconButton
-        key={`task_${id}`}
-        aria-label="task icon"
-        onClick={() => {
-          console.log({ id, status });
-        }}
-      >
-        <TaskStatusIcon status={status} size={16} />
-      </IconButton>
-    ))}
-  </>
-);
+interface RenderTaskIconsProps {
+  tasks: {
+    id: string;
+    status: string;
+  }[];
+}
+
+const RenderTaskIcons: React.FC<RenderTaskIconsProps> = ({ tasks }) => {
+  const { push } = useHistory();
+  return (
+    <>
+      {tasks.map(({ id, status }) => (
+        <IconButton
+          key={`task_${id}`}
+          aria-label="task icon"
+          onClick={() => push(getTaskRoute(id))}
+        >
+          <TaskStatusIcon status={status} size={16} />
+        </IconButton>
```

Comment:
We should wrap this with a Router link here instead of pushing to the history. 
Using a link allows us to be more declarative with our links and makes the links more accessible. Since they will be a part of the dom as opposed to something that programmatically changes the browser history. 
It also allows for easy usability use cases like being able to right click on a task icon and copy its link.